frontend/src/components/match/MatchCard.js

```diff
@@ -637,118 +699,36 @@ const TournamentMatch = props => {
           >
             <p class="text-center text-sm">No scores have been submitted yet</p>
           </Show>
+
           <Show when={isMatchTeamAdmin()}>
-            <button
-              data-modal-target={"submit-score-modal" + props.match?.id}
-              data-modal-toggle={"submit-score-modal" + props.match?.id}
-              type="button"
-              class={clsx(
-                "relative mb-2 mr-2 inline-flex items-center justify-center overflow-hidden rounded-lg p-0.5 font-medium",
-                "text-xs text-gray-900 hover:text-white focus:outline-none focus:ring-4 dark:text-white",
+            <SubmitScoreModal
+              buttonColor={
                 props.buttonColor
                   ? matchCardColorToButtonStyles[props.buttonColor]
                   : matchCardColorToButtonStyles[getMatchCardColor(props.match)]
-              )}
-            >
-              <Show
-                when={
-                  (isTeamAdminOf(props.match["team_1"].id) &&
-                    !props.match["suggested_score_team_1"]) ||
-                  (isTeamAdminOf(props.match["team_2"].id) &&
-                    !props.match["suggested_score_team_2"])
-                }
-                fallback={
-                  <>
-                    <span class="relative inline-flex items-center rounded-md bg-white px-3 py-2.5 transition-all duration-75 ease-in group-hover:bg-opacity-0 dark:bg-gray-800">
-                      Edit Score
-                      <Icon path={pencil} class="ml-1.5 w-4" />
-                    </span>
-                  </>
-                }
-              >
-                <span class="relative inline-flex items-center rounded-md bg-white px-3 py-2.5 transition-all duration-75 ease-in group-hover:bg-opacity-0 dark:bg-gray-800">
-                  Submit Score
-                  <Icon path={arrowRight} class="ml-1.5 w-4" />
-                </span>
-              </Show>
-            </button>
-          </Show>
-
-          {/*Submit Score modal*/}
-          <Show when={isMatchTeamAdmin()}>
-            <div
-              id={"submit-score-modal" + props.match?.id}
-              data-modal-backdrop="static"
-              tabIndex="-1"
-              aria-hidden="true"
-              class="fixed left-0 right-0 top-0 z-50 hidden h-[calc(100%-1rem)] max-h-full w-full overflow-y-auto overflow-x-hidden p-4 md:inset-0"
+              }
+              button={
+                hasUserSubmittedScores()
```

Comment:
baiscally need to rename to hasUserNotSubmittedScores

Reply:
yeah stupid :disappointed: